airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/split_secrets/JsonSecretsProcessorTest.java

```diff
@@ -425,4 +436,89 @@ public void doesNotCopySecrets_inNestedNonCombinationNodeWhenDestinationMissing(
     assertEquals(expected, copied);
   }
 
+  @Nested
+  class NoOpTest {
+
+    @BeforeEach
+    public void setup() {
+      processor = JsonSecretsProcessorFactory.builder()
+          .copySecrets(false)
+          .maskSecrets(false)
+          .build()
+          .createJsonSecretsProcessor();
+    }
+
+    @Test
+    public void testCopySecrets() {
+      final JsonNode src = Jsons.jsonNode(ImmutableMap.builder()
+          .put("field1", "value1")
+          .put("field2", 2)
+          .put("additional_field", "dont_copy_me")
+          .put("secret1", "donttellanyone")
+          .put("secret2", "updateme")
+          .build());
+
+      final JsonNode dst = Jsons.jsonNode(ImmutableMap.builder()
+          .put("field1", "value1")
+          .put("field2", 2)
+          .put("secret1", JsonSecretsProcessor.SECRETS_MASK)
+          .put("secret2", "newvalue")
+          .build());
+
+      final JsonNode actual = processor.copySecrets(src, dst, SCHEMA_ONE_LAYER);
+
+      final JsonNode expected = Jsons.jsonNode(ImmutableMap.builder()
+          .put("field1", "value1")
+          .put("field2", 2)
+          .put("additional_field", "dont_copy_me")
+          .put("secret1", "donttellanyone")
+          .put("secret2", "updateme")
+          .build());
+
+      assertEquals(expected, actual);
+    }
+
+    private static Stream<Arguments> scenarioProvider() {
+      return Stream.of(
+          Arguments.of("array", true),
+          Arguments.of("array", false),
+          Arguments.of("array_of_oneof", true),
+          Arguments.of("array_of_oneof", false),
+          Arguments.of("nested_object", true),
+          Arguments.of("nested_object", false),
+          Arguments.of("nested_oneof", true),
+          Arguments.of("nested_oneof", false),
+          Arguments.of("oneof", true),
+          Arguments.of("oneof", false),
+          Arguments.of("optional_password", true),
+          Arguments.of("optional_password", false),
+          Arguments.of("postgres_ssh_key", true),
+          Arguments.of("postgres_ssh_key", false),
+          Arguments.of("simple", true),
+          Arguments.of("simple", false));
+    }
+
+    @ParameterizedTest
+    @MethodSource("scenarioProvider")
+    void testSecretScenario(final String folder, final boolean partial) throws IOException {
+      final ObjectMapper objectMapper = new ObjectMapper();
+
+      final InputStream specIs = getClass().getClassLoader().getResourceAsStream(folder + "/spec.json");
+      final JsonNode specs = objectMapper.readTree(specIs);
+
+      final String inputFilePath = folder + (partial ? "/partial_config.json" : "/full_config.json");
+      final InputStream inputIs = getClass().getClassLoader().getResourceAsStream(inputFilePath);
+      final JsonNode input = objectMapper.readTree(inputIs);
+
+      final String expectedFilePath = folder + (partial ? "/partial_config.json" : "/full_config.json");
```

Comment:
is this meant to be `/updated_partial_config.json` and `/updated_config.json`?

also, how does `expected.json` fit into this test?

Reply:
It is expected that we do not modify anything here, on line 445 we set the maskScecrets and copy Secrets as false.